x/mongo/driver/compression.go

```diff
@@ -41,6 +40,50 @@ func getZstdEncoder(l zstd.EncoderLevel) (*zstd.Encoder, error) {
 	return encoder, nil
 }
 
+var zlibEncoders = &sync.Map{}
+
+func getZlibEncoder(l int) (*zlibEncoder, error) {
+	if v, ok := zlibEncoders.Load(l); ok {
+		return v.(*zlibEncoder), nil
+	}
+	b := bytes.NewBuffer(nil)
```

Comment:
Optional way of constructing a new bytes.Buffer without the awkward "nil" initializer.
```suggestion
	b := new(bytes.Buffer) 
```